### PR TITLE
Adjust SLIC*_ROOT* macros to automatically retrieve rank

### LINKED ISSUE #400: Enable SLIC_ERROR_ROOT without a rank argument
We have a bunch of SLIC_ERROR and SLIC_ERROR_IF calls that should really be SLIC_ERROR_ROOT. This is because we often don't have clean access to the MPI rank. Would it be possible to  not require the rank as an argument? I know @white238 looked into this before and thought it would be messy, but I see this becoming a fairly large problem in terms of verbosity of output on large runs.

### PR DESCRIPTION
**Pull request by joshessman-llnl**
Resolves #400

### CODE REVIEW

File: tests/serac_input.cpp [hunk 1]
```diff
@@ -42,7 +42,7 @@ namespace serac {
 TEST_F(InputTest, vec_1d)
 {
   reader_->parseString("vec = { x = 4.75 }");
-  auto& vec_table = inlet_->addTable("vec");
+  auto& vec_table = inlet_->addStruct("vec");
```
Comment (jamiebramwell) [inline tests/serac_input.cpp:45]: oh no! I think I ran my sed script on `src` only! Thanks for catching this.

**Review by jamiebramwell — APPROVED** [review]
This will clean things up nicely! Thanks!

File: tests/serac_input.cpp (hunk 1 above)
Comment (joshessman-llnl) [reply tests/serac_input.cpp:45]: I'm using a new Axom version so I've had to fix it in all of my branches (e.g. #398) to even get it to compile 😆

Comment (white238) [reply tests/serac_input.cpp:45]: This does keep happening.. should we move unit tests under src next to their units when we move to our more strict integration/unit tests split?

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
LGTM

**Pull request merged**